### PR TITLE
Update README for aggregation and cleanup V5

### PR DESCRIPTION
**Pull request by hush-hush**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +130,60 @@ After the client is created, you can start sending events to your Datadog Event
...
 After the client is created, you can start sending Service Checks to Datadog. See the dedicated [Service Check Submission: DogStatsD documentation](https://docs.datadoghq.com/developers/service_checks/dogstatsd_service_checks_submission/?code-lang=go) to see how to submit a Service Check to Datadog.
 
+## Client side aggregation
+
+Starting with version `5.0.0` (and `3.6.0` in beta), the client offers aggregation or value packing on the client side.
+
+This feature aims at reducing both the number of packets sent to the Agent and the packet drops in very high throughput
+scenarios.
+
+The aggregation window is 2s by default and can be changed through `WithAggregationInterval()` options. Note that the
```
Comment (vickenty) [inline README.md:140]:
```suggestion
The aggregation window is 2s by default and can be changed through `WithAggregationInterval()` option. Note that the
```

File: README.md [hunk 2]
```diff
@@ -103,6 +130,60 @@ After the client is created, you can start sending events to your Datadog Event
...
+aggregation window on the Agent side is 10s for DogStatsD metrics. So for example, setting a aggregation window of 3s on
```
Comment (vickenty) [inline README.md:141]:
```suggestion
aggregation window on the Agent side is 10s for DogStatsD metrics. So for example, setting an aggregation window of 3s in
```

File: README.md [hunk 3]
```diff
@@ -103,6 +130,60 @@ After the client is created, you can start sending events to your Datadog Event
...
+New metrics `datadog.dogstatsd.client.aggregated_context` and `datadog.dogstatsd.client.aggregated_context_by_type` have
+been introduced. See the [Monitoring this client](#monitoring-this-client) section.
+
+### "Basic" aggregation
+
+Enabled by default, the client will aggregate `gauge, `count` and `set`.
+
+### "Extended" aggregation
+
+This feature is only compatible with Agent's version >=6.25.0 && <7.0.0 or Agent's versions >=7.25.0.
```
Comment (vickenty) [inline README.md:157]:
```suggestion
This feature is only compatible with Agent versions starting from 6.25.0 and 7.25.0.
```

File: README.md [hunk 4]
```diff
@@ -103,6 +130,60 @@ After the client is created, you can start sending events to your Datadog Event
...
+
+Disabled by default, the client can also pack multiple values for `histogram`, `distribution` and `timing` in one
+message. Real aggregation is not possible for those types since the Agent also aggregates and two aggregation levels
+would change the final value sent to Datadog.
+
+When this option is enabled, the agent will buffer the metrics by context (unique combination of metrics name and tag
+set), and flush them in the fewest number of messages.
```
Comment (vickenty) [inline README.md:164]:
```suggestion
When this option is enabled, the agent will buffer the metrics by combination of metric name and tags, and send them in the fewest number of messages.
```

File: README.md (hunk 3 above)
Comment (hush-hush) [reply README.md:157]: Actually this is the wording we find with documentation since 7.24 is not compatible for example.

**Pull request merged**